salt/states/file.py

```diff
@@ -6312,7 +6315,11 @@ def serialize(name,
                         'result': False}
 
             with salt.utils.files.fopen(name, 'r') as fhr:
-                existing_data = __serializers__[deserializer_name](fhr, **options.get(serializer_name, {}))
+                try:
+                    existing_data = __serializers__[deserializer_name](fhr, **options.get(serializer_name, {}))
```

Comment:
instead of just passing in all the args, i would do something like.

```
goodopts = salt.utils.args.get_function_argspec(__serializers__[deserializer_name]).args
opts = opts.get(serializer_name, {})
for opt in list(opts):
    if opt not in goodopts:
        opts.pop(opt, None)
```

And only pass in the key word arguments if the deserializer actually accepts those arguments.

Reply:
That was my first instinct but this is not the inability of serializer/deserializer to accept args, but actual python stdlib JSONDecode doesn't accept it. I don't know how could I test if some random deserializer will be able to accept our kwargs except in a try block. Do you have an idea if that's possible?

Reply:
ahh yes, in that case, this is the best option.  But it would be good to add a debug log that says that this exception was caught and that the function is being run without the options.

Reply:
Added debug output that we caught the exception with what options were passed to it.